File: Milan Das/Day-03 (07-07-2023)/Diamond.java
package com.algorithm;

/*
	b)  Draw a diamond style of asterisks?
					        *        
					       ***       
					      *****      
					     *******     
					    *********    
					   ***********   
					  *************  
					 *************** 
					*****************
					 *************** 
					  *************  
					   ***********   
					    *********    
					     *******     
					      *****      
					       ***       
					        *        
 */

public class Diamond {

	public static void main(String[] args) {
		int spaces = 40;
		int stars = 1;
		
		for(int i=1; i<=17; i++) {
			for(int j=1; j<=spaces; j++) System.out.print(" ");
			for(int j=1; j<=stars; j++)	 System.out.print("*");
			System.out.println();
			if(i<9) {
				spaces--; stars+=2;
			} else {
				spaces++; stars-=2;
			}
		}
		
	}

}

File: Assignment/Saha Sahil/Day-10(18-072023)/diamond.py
 WAP to draw a diamond style of asterisks?
  

                                     *           
                                    ***                
                                   *****                 
                                  *******            
                                 *********             
                                ***********           
                               *************         
                              *************** 
                             *****************     
                              ***************                         
                               *************
                                ***********
                                 *********
                                  *******
                                   *****
                                    ***
                                     *
def draw_diamond(rows):
    for i in range(rows):
        print(' ' * (rows - i - 1) + '*' * (2 * i + 1))
    for i in range(rows - 2, -1, -1):
        print(' ' * (rows - i - 1) + '*' * (2 * i + 1))
num_rows = int(input("Enter the number of rows for the diamond: "))
draw_diamond(num_rows)
